tests/patmat/i12337.scala

```diff
@@ -0,0 +1,12 @@
+sealed trait Status
+object Status {
+  case object Active   extends Status
+  case object Inactive extends Status
```

Comment:
Can one of these `case object` be `case class` instead to make sure it works for `case class` as well?

e.g.)
```scala
sealed trait Status
object Status {
  case class Active(since: Int) extends Status
  case object Inactive extends Status
}

case class Foo(status: Status)
def bar(user: Foo): Unit = user match {
  case Foo(Status.Active(2000)) =>
    println(s"active since 2000")
}
// Expected:
// warning: match may not be exhaustive.
// It would fail on the following inputs: Foo(Active((x: Int forSome x not in 2000))), Foo(Inactive)
// def bar(user: Foo): Unit = user match {
```